final_task/rss_reader/news_articles.py

```diff
@@ -24,5 +24,16 @@ def __repr__(self):
         if self.news_description:
             result += f"{self.news_description}\n\n\n"
         result += links + '*'*80
-
         return result
+
+    def __key(self):
+        return (self.news_outlett_name, self.news_title, self.pub_date,
+                self.news_link, self.news_description, self.img_alt, self.img_src)
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, other):
+        if isinstance(other, NewsArticle):
+            return self.__key() == other.__key()
+        return NotImplemented
```

Comment:
A question: Why do you need to return this here?

Reply:
Initially I find it on stackoverflow as a good practice to implement __hash__(). In docs I found out that "If class does not define an __eq__() method it should not define a __hash__() operation either". So I decided to implement __hash__() with __eq__() together. And about "return NotImplemented" I found out that if the other object, we are comparing with, is not an instance of NewsArticle class, then this method goes back to built-it behavior.
Actually, for my project it is not necessary, so is it better I remove it? 
p.s. Am I get it all right?